### PR TITLE
Implement ArmPolling as subclass of msrest.polling.PollingMethod

### PR DESCRIPTION
**Pull request by lmazuel**
Needs https://github.com/Azure/msrest-for-python/pull/78

### CODE REVIEW

File: msrestazure/polling/arm_polling.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+            return self.async_url
+        elif self.location_url:
+            return self.location_url
+        elif self.method == "PUT":
+            return self.initial_response.request.url
+        else:
+            raise BadResponse("Unable to find a valid status link for polling")
+
+
+class ARMPolling(object):
```
Comment (annatisch) [inline msrestazure/polling/arm_polling.py]: Shouldn't this inherit from msrest.polling.PollingMethod?

File: msrestazure/polling/arm_polling.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    else:
+        return url
+
+
+class LongRunningOperation(object):
+    """LongRunningOperation
+    Provides default logic for interpreting operation responses
+    and status updates.
+    """
+    _convert = re.compile('([a-z0-9])([A-Z])')
```
Comment (annatisch) [inline msrestazure/polling/arm_polling.py]: This doesn't appear to be used anywhere?

File: msrestazure/polling/arm_polling.py (hunk 1 above)
Comment (lmazuel) [reply msrestazure/polling/arm_polling.py]: Indeed!

File: msrestazure/polling/arm_polling.py (hunk 2 above)
Comment (lmazuel) [reply msrestazure/polling/arm_polling.py]: That's a good point, even in the current code it's not...

**Comment by lmazuel** [comment]
@annatisch FYI last commit was translation of the AzureOperationPoller tests to LROPoller + ARMPolling

**Pull request merged**